src/parser/parser.yy

```diff
@@ -939,7 +939,12 @@ vertex_prop_expression
 
 var_prop_expression
     : VARIABLE DOT name_label {
-        $$ = VariablePropertyExpression::make(qctx->objPool(), *$1, *$3);
+        if (!qctx->existParameter(*$1)) {
+            $$ = VariablePropertyExpression::make(qctx->objPool(), *$1, *$3);
+        } else {
+            $$ = AttributeExpression::make(qctx->objPool(),
```

Comment:
Why need `AttributeExpression` here?

Reply:
In order to achieve expression like `map.k1.k2.k3`. 
IMO, `VariablePropertyExpression` can also be implemented as `VariableExpression`+`AttributeExpression`.